File: Human_written_code/Code_without_comments/Python/Q36.py
def main():
    t = int(input())
    for _ in range(t):
        res = 0
        n = int(input())
        arr = list(map(int, input().split()))
        
        count_map = {}
        for num in arr:
            count_map[num] = count_map.get(num, 0) + 1

        ans = max(count_map.values())
        
        while ans < n:
            if 2 * ans > n:
                res += 1 + (n - ans)
                ans = n
            else:
                res += 1 + ans
                ans *= 2

        print(res)

if __name__ == "__main__":
    main()

File: Human_written_code/Code_without_comments/CPP/Q36.cpp
#include <iostream>
#include<bits/stdc++.h>
using namespace std;
int main()
{
    long long t;
    cin>>t;
    while(t--)
    {
        long long res=0;
        long long n;
        cin>>n;
        long long arr[n];
        for(int i=0;i<n;i++)
        {
            cin>>arr[i];
        }
        long long cnt=1,ans=0;
        
        map<long long,long long> m;
        for(int i=0;i<n;i++)
        {
            m[arr[i]]++;
        }
        
        for(auto it:m)
        {
            ans = max(ans,it.second);
        }
        
        while(ans<n)
        {
            if(2*ans > n)
            {
                res++;
                res += (n-ans);
                ans=n;
            }
            else
            {
                res++;
                res += ans;
                ans = 2*ans;
            }
        }
        cout<<res<<endl;
    }
 
    
}
